File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int ans = 0, zero = 0, one;
        for(int i=0; i<s.size()-1; i++){
            if(s[i] == '0') zero++;
            one = 0;
            for(int j=i+1; j<s.size(); j++){
                if(s[j] == '1'){
                    one++;
                }
            }
            ans = max(ans, zero+one);
        }
        return zero==s.size() ? ans-1 : ans;
    }
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution {
    public int maxScore(String s) {
        int ans= 0 , one = 0, zero = 0;
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == '1') one++;
        }
        for(int i=0; i<s.length()-1; i++){
            if(s.charAt(i) == '0') zero++;
            else one--;
            ans = Math.max(ans, one+zero);
        }
        return ans;
    }
}